lib/Makefile

```diff
@@ -44,7 +44,12 @@ F90FLAGS ?=
 default: psy_data_base.o
 process: psy_data_base.f90
 
-.PHONY: default process clean
+.PHONY: default process clean all
+
+all:
+	$(MAKE) -C extract all
+	$(MAKE) -C profiling all
```

Comment:
It's not only that ... dl_timer, drhook, nvidia. I can't makeup my mind - being able to do 'make all' is convenient in some cases, but not in others. Maybe have two targets? ``make all_local`` and ``make all``? The first one only making files that are included in PSyclone?

Reply:
Yes, I like the idea of 2 targets (although it's very unlikely that anyone will ever have everything installed for 'all' - e.g. I've not got DrHook and you probably haven't got nvtx? - if so maybe we drop 'all' or limit it so that it's not actually *all* :-) )

Reply:
I've added a comment, so 'all' is only ``template`` and ``simple_timer``.